evalml/pipelines/components/transformers/imputers/simple_imputer.py

```diff
@@ -47,6 +47,10 @@ def fit(self, X, y=None):
         """
         X = _convert_to_woodwork_structure(X)
         X = _convert_woodwork_types_wrapper(X.to_dataframe())
+
+        if (X.dtypes == bool).all():
```

Comment:
[Suggested alternative:](https://pandas.pydata.org/pandas-docs/version/0.25.3/reference/api/pandas.api.types.is_bool_dtype.html) `pd.ptypes.is_bool_dtype(X).all()`

Reply:
It might not have the same behavior, as it checks the individual data type rather than the series/dataframe type. As such, this would return True for types that are able to be handled by the imputer just fine (such as category). 

![image](https://user-images.githubusercontent.com/8752455/100254411-45163e80-2f10-11eb-9cd4-03b7ef92a212.png)